state/executor.go

```diff
@@ -272,17 +273,6 @@ func (t *Transition) WriteFailedReceipt(txn *types.Transaction) error {
 
 // Write writes another transaction to the executor
 func (t *Transition) Write(txn *types.Transaction) error {
-	signer := crypto.NewSigner(t.config, uint64(t.r.config.ChainID))
-
-	var err error
-	if txn.From == emptyFrom {
-		// Decrypt the from address
-		txn.From, err = signer.Sender(txn)
-		if err != nil {
-			return NewTransitionApplicationError(err, false)
-		}
-	}
```

Comment:
Is it ok to move this part from here to `apply` (i.e. [checkAndProcessLegacyTx](https://github.com/0xPolygon/polygon-edge/pull/809/files#diff-f64dae91a36980ba02c538f69fb83b7c80a8b4345db96849127307d33ccfacd1R843-R852)? For this reason, `TestFetchBLSValidators` and `TestFetchECDSAValidators` started failing.
It is either:
1. leave it as it was before (check sender in `Write` function)
2. introduce this change and set `Transaction.From` field to valid address

Reply:
fixed